src/Symfony/Component/String/AbstractString.php

```diff
@@ -0,0 +1,555 @@
+                    if ($lastError === $v && '_ERROR' === substr($k, -6)) {
+                        throw new RuntimeException('Matching failed with '.$k.'.');
+                    }
+                }
+
+                throw new RuntimeException('Matching failed with unknown error code.');
+            }
+        } finally {
+            restore_error_handler();
+        }
+
+        $str = clone $this;
+        $str->string = $string;
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    abstract public function slice(int $start = 0, int $length = null): self;
+
+    /**
+     * @return static
+     */
+    abstract public function splice(string $replacement, int $start = 0, int $length = null): self;
+
+    /**
+     * @return static[]
+     */
+    abstract public function split(string $delimiter, int $limit = null): array;
+
+    abstract public function startsWith(string $prefix): bool;
+
+    /**
+     * @return static
+     */
+    public function title(bool $allWords = false): self
+    {
+        $str = clone $this;
+
+        if ($allWords) {
+            $str->string = preg_replace_callback('/\b./u', static function ($m) {
+                return mb_convert_case($m[0], MB_CASE_TITLE, 'UTF-8');
+            }, $str->string);
+        } else {
+            $firstChar = mb_substr($str->string, 0, 1, 'UTF-8');
+            $str->string = mb_convert_case($firstChar, MB_CASE_TITLE, 'UTF-8').substr($str->string, \strlen($firstChar));
+        }
+
+        return $str;
+    }
+
+    public function toBinary(): BinaryString
+    {
+        return new BinaryString($this->string, $this->locale);
+    }
+
+    public function toGrapheme(): GraphemeString
+    {
+        return new GraphemeString($this->string, $this->locale);
+    }
+
+    public function toUtf8(): Utf8String
+    {
+        return new Utf8String($this->string, $this->locale);
+    }
+
+    /**
+     * @return static
+     */
+    public function trim(string $chars = " \t\n\r\0\x0B\x0C\u{A0}\u{FEFF}"): self
+    {
+        if (!preg_match('//u', $chars)) {
+            throw new InvalidArgumentException('Invalid UTF-8 chars.');
+        }
+        $chars = preg_quote($chars);
+
+        $str = clone $this;
+        $str->string = preg_replace("{^[$chars]++|[$chars]++$}u", '', $str->string);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function trimEnd(string $chars = " \t\n\r\0\x0B\x0C\u{A0}\u{FEFF}"): self
+    {
+        if (!preg_match('//u', $chars)) {
+            throw new InvalidArgumentException('Invalid UTF-8 chars.');
+        }
+        $chars = preg_quote($chars);
+
+        $str = clone $this;
+        $str->string = preg_replace("{[$chars]++$}u", '', $str->string);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function trimStart(string $chars = " \t\n\r\0\x0B\x0C\u{A0}\u{FEFF}"): self
```

Comment:
` \t\n\r\0\x0B\x0C\u{A0}\u{FEFF}` could be defined as class constant and used as `$chars`'s default value. This exact chars list is used 3 times in different `trim` methods.